DiscordChatExporter.Domain/Exporting/MediaDownloader.cs

```diff
@@ -1,44 +1,72 @@
 ﻿using System;
 using System.Collections.Generic;
 using System.IO;
+using System.Net;
 using System.Net.Http;
 using System.Text.RegularExpressions;
 using System.Threading.Tasks;
 using DiscordChatExporter.Domain.Internal;
 using DiscordChatExporter.Domain.Internal.Extensions;
+using Polly;
+using Polly.Retry;
 
 namespace DiscordChatExporter.Domain.Exporting
 {
     internal partial class MediaDownloader
     {
         private readonly HttpClient _httpClient = Singleton.HttpClient;
         private readonly string _workingDirPath;
+        private readonly AsyncRetryPolicy _httpRequestPolicy;
 
         private readonly Dictionary<string, string> _pathMap = new Dictionary<string, string>();
 
         public MediaDownloader(string workingDirPath)
         {
             _workingDirPath = workingDirPath;
+
+            _httpRequestPolicy = Policy
+                .Handle<IOException>()
+                .WaitAndRetryAsync(GenerateSleepDurations());
```

Comment:
Can we just simplify to something like:

```csharp
.WaitAndRetryAsync(6, i => TimeSpan.FromSeconds(0.5*i))
```

I realize your strategy is more advanced, but honestly my brain hurts trying to understand it :D

Reply:
Yeah the policy I put in was a bit silly--it's more so meant to deal with multiple threads competing over a single resource, which is completely unrelated to what we are doing. It's just wired into my head as my go-to whenever I hear "retry." I did what you suggested, but upped the retry count to 8 to give a total wait time of 18 seconds rather than 10.5.